docs/developer_guide.md

```diff
@@ -96,8 +96,7 @@ pip install -r requirements.txt -e sharktank/ shortfin/
 # Optionally clone and install the latest editable iree-turbine dep in deps/,
 # along with nightly versions of iree-base-compiler and iree-base-runtime.
 pip install -f https://iree.dev/pip-release-links.html --upgrade --pre \
-  iree-base-compiler iree-base-runtime --src deps \
-  -e "git+https://github.com/iree-org/iree-turbine.git#egg=iree-turbine"
+  iree-base-compiler iree-base-runtime iree-turbine
```

Comment:
Update the comment to match the code. I think for a developer guide we should mention the editable source install too, but still recommend most people to start with nightly packages.

Reply:
I did but missed to save and commit -.-

Reply:
I restored / added a comment pointing to the editable iree-turbine.